File: app/src/main/java/me/larikraun/room_devfestsw/data/AppDatabase.java
package me.larikraun.room_devfestsw.data;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.RoomDatabase;
import android.arch.persistence.room.migration.Migration;
import android.support.annotation.NonNull;

/**
 * Author: Omolara Adejuwon
 * Date: 11/8/17.
 */
@Database(entities = {Attendee.class}, version = 1)

public abstract class AppDatabase extends RoomDatabase {
	public abstract AttendeeDao attendeeDao ();
	
	public static final Migration MIGRATION_1_2 = new Migration (1, 2) {
		
		@Override
		public void migrate (@NonNull SupportSQLiteDatabase database) {
			database.execSQL ("ALTER TABLE attendee ADD COLUMN phone_number TEXT");
		}
	};
}

File: app/src/main/java/me/larikraun/room_devfestsw/data/AppDatabase.kt
package me.larikraun.room_devfestsw.data

import android.arch.persistence.db.SupportSQLiteDatabase
import android.arch.persistence.room.Database
import android.arch.persistence.room.RoomDatabase
import android.arch.persistence.room.migration.Migration


/**
 * Author: Omolara Adejuwon
 * Date: 11/8/17.
 */
@Database(entities = arrayOf(Attendee::class), version = 2)

abstract class AppDatabase : RoomDatabase() {
    abstract fun attendeeDao(): AttendeeDao

    companion object {

        val MIGRATION_1_2: Migration = object : Migration(1, 2) {
            override fun migrate(database: SupportSQLiteDatabase) {
                database.execSQL("ALTER TABLE attendee ADD COLUMN phone_number TEXT NOT NULL DEFAULT nothing")
            }
        }
    }
}